src/main/java/com/google/devtools/build/lib/starlarkbuildapi/CommandLineArgsApi.java

```diff
@@ -66,6 +66,14 @@
             + "<li>Values that are already strings are left as-is." //
             + "<li><a href='../builtins/File.html'><code>File</code></a> objects are turned into"
             + " their     <code>File.path</code> values." //
+            + "<li><a href='../builtins/Label.html'><code>Label</code></a> objects are turned into"
+            + " a string representation that resolves back to the same object when resolved in the"
+            + " context of the main repository. If possible, the string representation uses the"
+            + " apparent name of a repository in favor of the repository's canonical name, which"
+            + " makes this representation suited for use in BUILD files. While the exact form of"
+            + " the representation is not guaranteed, typical examples are"
```

Comment:
Why is it not guaranteed?

Reply:
I didn't want to commit to the full details of how labels are formed: Are we emitting `@foo` as `@foo` or `@foo//:foo`? Do we always return labels with an apparent repo name if there is one for the repo? I could expand on the guarantees if you think that's worthwhile and doesn't unnecessarily constrain us.

Reply:
I think we should get an opinion from @Wyverald for this documentation. I don't deal with repository mappings much so I'm not sure what to document here. In general documentation that makes it sound like a mystery is concerning to me, so an additional opinion would help.

Reply:
I don't have strong opinions here; but IMO the things you mentioned --

> Are we emitting `@foo` as `@foo` or `@foo//:foo`? Do we always return labels with an apparent repo name if there is one for the repo?

-- are perfectly fine to guarantee. I can't really foresee a situation where we'd want to emit `@foo//:foo` instead of `@foo`, or use a canonical label when an apparent one exists.

Reply:
We do emit `@foo//:foo` at the moment though. I could see getting rid of the colon ending up breaking tooling that consumes these labels, so I'm not sure whether that would make it safe to cherry-pick. At the same time, guaranteeing that we never use the shorthand also doesn't seem ideal...

Reply:
hmm, that's fair. I'll leave it up to your judgment.

Reply:
Since the change would be doc only, I'll mentally snooze and revisit it later.